modules/andes-core/client/src/main/java/org/wso2/andes/client/BasicMessageConsumer.java

```diff
@@ -619,7 +619,9 @@ public void close(boolean sendClose) throws JMSException
                         // no point otherwise as the connection will be gone
                         if (!_session.isClosed() || _session.isClosing())
                         {
-                            _session._dispatcher.rejectPending(this);
+                            if(null != _session._dispatcher) {
```

Comment:
The reject call was brought here from AMQSession since we wanted to send reject call before the session close otherwise a NullPointer will occur in the server side.

https://github.com/wso2/andes/commit/3f39004c96980ad2578f0f5faacda59eb587f6f2